core/logger/logger.go

```diff
@@ -13,15 +13,57 @@ import (
 	"io"
 	"log"
 	"os"
+	"time"
 
 	"github.com/fatih/color"
+	"github.com/getsentry/sentry-go"
+	"github.com/smartcontractkit/chainlink/core/static"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+const SentryFlushDeadline = 5 * time.Second
+
 var envLvl = zapcore.InfoLevel
 
 func init() {
+	// If SENTRY_ENVIRONMENT is set, it will override everything. Otherwise infers from CHAINLINK_DEV.
+	var sentryenv string
+	if env := os.Getenv("SENTRY_ENVIRONMENT"); env != "" {
+		sentryenv = env
+	} else if os.Getenv("CHAINLINK_DEV") == "true" {
+		sentryenv = "dev"
+	} else {
+		sentryenv = "prod"
+	}
+	// If SENTRY_DSN is set, it will override everything. Otherwise static.SentryDSN will be used.
+	// If neither are set, sentry is disabled.
+	var sentrydsn string
+	if dsn := os.Getenv("SENTRY_DSN"); dsn != "" {
+		sentrydsn = dsn
+	} else {
+		sentrydsn = static.SentryDSN
+	}
+	// If SENTRY_RELEASE is set, it will override everything. Otherwise, static.Version will be used.
+	var sentryrelease string
+	if release := os.Getenv("SENTRY_RELEASE"); release != "" {
+		sentryrelease = release
+	} else {
+		sentryrelease = static.Version
+	}
+	err := sentry.Init(sentry.ClientOptions{
+		// AttachStacktrace is needed to send stacktrace alongside panics
+		AttachStacktrace: true,
+		Dsn:              sentrydsn,
+		Environment:      sentryenv,
+		Release:          sentryrelease,
+		// Enable printing of SDK debug messages.
+		// Uncomment line below to debug sentry
+		// Debug: true,
+	})
+	if err != nil {
+		log.Fatalf("sentry.Init: %s", err)
+	}
```

Comment:
nit: It would be nice not to separate `envLvl`'s declaration before this block from assignment below. Maybe this deserves it's own helper func or even separate sentry.go file?